admiral/pkg/clusters/serviceentry.go

```diff
@@ -309,16 +309,29 @@ func generateProxyVirtualServiceForDependencies(ctx context.Context, remoteRegis
 	return nil
 }
 
-func createOrUpdateVirtualService(ctx context.Context, remoteController *RemoteController, virtualService *v1alpha3.VirtualService) error {
-	existingVS, err := remoteController.VirtualServiceController.IstioClient.NetworkingV1alpha3().VirtualServices(common.GetSyncNamespace()).Get(ctx, virtualService.Name, v12.GetOptions{})
-	if err != nil && k8errors.IsNotFound(err) {
-		log.Infof("virtualService %s not found", virtualService.Name)
-	}
-	err = addUpdateVirtualService(ctx, virtualService, existingVS, common.GetSyncNamespace(), remoteController)
+func getAdmiralGeneratedVirtualService(ctx context.Context, remoteController *RemoteController, listOptions v12.ListOptions,
+	namespace string) (*v1alpha3.VirtualService, error) {
+	existingVSList, err := remoteController.VirtualServiceController.IstioClient.NetworkingV1alpha3().
+		VirtualServices(namespace).List(ctx, listOptions)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	return nil
+	if existingVSList == nil {
+		return nil, fmt.Errorf("error fetching virtualservice with labels %s", listOptions.LabelSelector)
+	}
+	if len(existingVSList.Items) == 0 {
+		return nil, fmt.Errorf("no virtualservice found with labels %s", listOptions.LabelSelector)
+	}
+	var result *v1alpha3.VirtualService
+	for _, existingVS := range existingVSList.Items {
+		if isGeneratedByAdmiral(existingVS.Annotations) {
+			result = existingVS
+		}
+	}
+	if result == nil {
+		return nil, fmt.Errorf("no virtualservice found with labels %s and annotation %s=%s", listOptions.LabelSelector, resourceCreatedByAnnotationLabel, resourceCreatedByAnnotationValue)
```

Comment:
Is this a valid scenario where there are no VS created for dependency proxy  in a given cluster?

Reply:
@nirvanagit , this is independent of DependencyProxy. This is the case where the services are part of mesh and communication is not being proxied through a proxy mesh service.